### PR TITLE
Don't deduplicate failed submissions

### LINKED ISSUE #5336: Command deduplication: remove entry after a failed submission
After a command fails at submission, remove the corresponding entry from the deduplication cache.

### PR DESCRIPTION
**Pull request by rautenrieth-da**
This PR removes commands from the deduplication cache if the *submission* is rejected.

Removing commands on rejected transactions will follow.

Fixes #5336.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala [hunk 1]
```diff
@@ -1693,6 +1702,22 @@ private class JdbcLedgerDao(
...
 
+  private val SQL_DELETE_COMMAND = SQL("""
+     |delete from participant_command_submissions
+     |where deduplicate_key < {deduplicationKey}
+    """.stripMargin)
+
+  override def stopDeduplicatingCommand(
+      commandId: domain.CommandId,
+      submitter: Party): Future[Unit] =
+    dbDispatcher.executeSql("remove_expired_deduplication_data") { implicit conn =>
```
Comment (stefanobaghino-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala:1713]: The description here looks like a better name for the method.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/LedgerDao.scala [hunk 2]
```diff
@@ -161,6 +164,18 @@ trait LedgerReadDao extends ReportsHealth {
   def removeExpiredDeduplicationData(
       currentTime: Instant,
   ): Future[Unit]
+
+  /**
+    * Stops deduplicating the given command.
```
Comment (stefanobaghino-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/LedgerDao.scala:169]: This does not add a lot of information. I would recommend adding some context or deleting the docstring.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala (hunk 1 above)
Comment (rautenrieth-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala:1713]: Oh, copy and paste error, there's a different method for removing expired deduplication data.

Comment (rautenrieth-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala:1713]: For reference, `stopDeduplicatingCommand` removes a deduplication entry prior to its expiry. Entries should be removed when the commands are rejected.

**Comment by gerolf-da** [comment]
@rautenrieth-da: Am I missing something, or is `stopDeduplicatingCommand` not being called for a failed transaction coming from the ledger? I would have expected changes in `JdbcIndexer` and `SqlLedger` for that.

**Comment by rautenrieth-da** [comment]
@gerolf-da No, I wanted to it in separate PRs, see the "Removing commands on rejected transactions will follow." in the PR description.

**Pull request merged**